packages/material-ui/src/ButtonBase/TouchRipple.test.js

```diff
@@ -1,88 +1,114 @@
 import React from 'react';
 import { useFakeTimers } from 'sinon';
-import { assert } from 'chai';
-import { createShallow, createMount, getClasses, unwrap } from '@material-ui/core/test-utils';
-import describeConformance from '../test-utils/describeConformance';
-import Ripple from './Ripple';
+import { expect } from 'chai';
+import { createMount, getClasses } from '@material-ui/core/test-utils';
+import { cleanup, createClientRender } from 'test/utils/createClientRender';
 import TouchRipple, { DELAY_RIPPLE } from './TouchRipple';
 
 const cb = () => {};
 
 describe('<TouchRipple />', () => {
-  let shallow;
   let mount;
-  let classes;
-  const TouchRippleNaked = unwrap(TouchRipple);
+  // StrictModeViolation: uses react-transition-group
+  const render = createClientRender({ strict: false });
+
+  /**
+   * @param {HTMLElement} container returned by render
+   * @param {number} active of active ripples to check for
+   * @param {number} stopping of stopping ripples to check for
+   */
+  function checkRipples(container, active, stopping) {
+    expect(
+      container.querySelectorAll('.ripple-visible .child:not(.child-leaving)'),
+    ).to.have.lengthOf(active);
+    expect(container.querySelectorAll('.ripple-visible .child-leaving')).to.have.length(stopping);
+  }
+
+  /**
+   * @param {object} other props to spread to TouchRipple
+   */
+  function createTouchRipple(other) {
+    const touchRippleRef = React.createRef();
+    const { container } = render(
+      <TouchRipple
+        ref={touchRippleRef}
+        classes={{
+          ripple: 'ripple',
+          rippleVisible: 'ripple-visible',
+          child: 'child',
+          childLeaving: 'child-leaving',
+        }}
+        {...other}
+      />,
+    );
+
+    const instance = touchRippleRef.current;
+    return { container, instance };
+  }
 
   before(() => {
-    shallow = createShallow({ dive: true });
-    mount = createMount({ strict: undefined });
-    classes = getClasses(<TouchRipple />);
+    // StrictModeViolation: uses react-transition-group
+    mount = createMount({ strict: false });
   });
 
   after(() => {
+    cleanup();
     mount.cleanUp();
   });
 
+/* Breaks due to ref being used for an imperative handle
```

Comment:
Thoughts? @eps1lon @oliviertassinari

Reply:
`refInstanceof: Object` should do or something similar.

Reply:
This kind of worked, what do you think of the changed I've made to parseTest and generateMarkdown? @eps1lon